cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/requestrate/RequestRateModule.java

```diff
@@ -62,20 +66,26 @@ public RequestRateModule(ModuleConfiguration configuration, List<Reporter> repor
         period = config.period();
         timeunit = config.timeunit();
         rateFactor = timeunit.toSeconds(period);
+        requestRates = initRequestRates(config);
 
-        logger.info("RequestRate module initialized with {} {} reporting period.", period, timeunit.name());
-        requestRates = new HashMap<>();
-        for (StatementType statementType : StatementType.values()) {
-            requestRates.put(statementType, new AtomicCounter());
-        }
+        logger.info("RequestRate module initialized with {} {} reporting period and requests to report: {}.", period,
+                timeunit.name(), requestRates.keySet().toString());
 
         timer = new Timer(REQUEST_RATE_THREAD_NAME);
         timer.schedule(new RequestRateTask(), 0, config.reportingRateInMillis());
     }
 
     @Override
     public void process(Query query) {
-        requestRates.get(query.statementType()).increment();
+        if (requestRates.containsKey(ALL_REQUESTS_TO_REPORT)) {
```

Comment:
But in case of `:` what will initial structure look like. I guess in a case of SELECT and UPDATE it will be map of two values and each statement type will have 12 consistency levels and you would report 24 measurements even though you are not using them? 

I wanted to have single measurement if you want to report all values, no matter which statement type and consistency level it is.